### PR TITLE
Add updates from iqm_client

### PR DESCRIPTION
**Pull request by rreiner-hqs**

### CODE REVIEW

File: roqoqo-iqm/src/backend.rs [hunk 1]
```diff
@@ -494,14 +553,11 @@ impl EvaluatingBackend for Backend {
...
 
-        #[derive(PartialEq, Debug, Clone, Serialize, Deserialize)]
-        struct ResponseBody {
-            id: String,
-        }
         let job_id: &str = &serde_json::from_str::<ResponseBody>(&resp.text().unwrap())
             .unwrap()
             .id;
 
+        println!("{:?}", self.abort_job(job_id));
```
Comment (kbarkhqs) [inline roqoqo-iqm/src/backend.rs:560]: Should this be removed?

File: roqoqo-iqm/src/backend.rs [hunk 2]
```diff
@@ -451,6 +469,47 @@ impl Backend {
...
+
+    /// Abort a submitted job.
+    ///
+    /// # Arguments
+    ///
+    /// `id` - The ID of the job to abort
+    ///
+    /// # Returns
+    ///
+    /// Err(RoqoqoBackendError) - If the job abortion failed.
```
Comment (kbarkhqs) [inline roqoqo-iqm/src/backend.rs:481]: missing the `` marks

File: roqoqo-iqm/src/backend.rs [hunk 3]
```diff
@@ -451,6 +469,47 @@ impl Backend {
...
+                let abort_failed_msg: &str =
+                    &serde_json::from_str::<AbortResponse>(&resp.text().unwrap())
+                        .unwrap()
+                        .detail;
+                Err(RoqoqoBackendError::GenericError {
+                    msg: format!("Job abortion failed: {}", abort_failed_msg),
+                })
+            }
+        }
+    }
```
Comment (kbarkhqs) [inline roqoqo-iqm/src/backend.rs:512]: Should there be an additional test to cover this functionality?

**Review by kbarkhqs — COMMENTED** [review]
Should there be an additional test for the new aborting function? Also, this addition should be documented in the CHANGELOG.

File: roqoqo-iqm/src/backend.rs (hunk 3 above)
Comment (nfwvogt) [reply roqoqo-iqm/src/backend.rs:512]: Since it's the live API it's hard to test that in CI.

File: roqoqo-iqm/src/backend.rs (hunk 1 above)
Comment (nfwvogt) [inline roqoqo-iqm/src/backend.rs:560]: This looks like a left-over debug statement to me.

**Review by kbarkhqs — APPROVED** [review]

**Pull request merged**